src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -205,7 +217,13 @@ public static partial class ExecutableResourceBuilderExtensions
 
         public static ApplicationModel.IResourceBuilder<ApplicationModel.ExecutableResource> AddExecutable(this IDistributedApplicationBuilder builder, string name, string command, string workingDirectory, params string[]? args) { throw null; }
 
-        public static ApplicationModel.IResourceBuilder<T> PublishAsDockerFile<T>(this ApplicationModel.IResourceBuilder<T> builder, System.Collections.Generic.IEnumerable<ApplicationModel.DockerBuildArg>? buildArgs = null)
+        public static ApplicationModel.IResourceBuilder<T> PublishAsDockerFile<T>(this ApplicationModel.IResourceBuilder<T> builder, System.Action<ApplicationModel.IResourceBuilder<ApplicationModel.ContainerResource>>? configure)
+            where T : ApplicationModel.ExecutableResource { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<T> PublishAsDockerFile<T>(this ApplicationModel.IResourceBuilder<T> builder, System.Collections.Generic.IEnumerable<ApplicationModel.DockerBuildArg>? buildArgs)
```

Comment:
We should deprecate this API.

Reply:
@davidfowl @joperezr - how do you want to track comments like these? Should we open issues when we agree in comments here?

Reply:
That is one, or the other would be to use each of these threads as tracking, so for instance, if we do agree to deprecate this then until the PR is merged to remove it we mark this thread as resolved. Before the release we could do a pass on this PR to make sure there aren't any unresolved threads. Thoughts? Basically want to try to optimize for efficiency, but of course this does mean that we don't have visibility of this work in our project board.

Reply:
https://github.com/dotnet/aspire/pull/7485